addons/stock/models/stock_quant.py

```diff
@@ -0,0 +1,497 @@
+        help="Technical field used to record the destination location of a move that created a negative quant")
+
+    @api.one
+    def _compute_name(self):
+        """ Forms complete name of location from parent location to child location. """
+        self.name = '%s: %s%s' % (self.lot_id.name or self.product_id.code or '', self.qty, self.product_id.uom_id.name)
+
+    @api.multi
+    def _compute_inventory_value(self):
+        for quant in self:
+            # TDE note: there was something about poping force_company -> usefull ?
+            if quant.company_id != self.env.user.company_id:
+                # if the company of the quant is different than the current user company, force the company in the context
+                # then re-do a browse to read the property fields for the good company.
+                quant = quant.with_context(force_company=quant.company__id.id)
+            quant.inventory_value = quant._get_inventory_value()
+
+    @api.multi
+    def _get_inventory_value(self):
+        self.ensure_one()
+        return self.product_id.standard_price * self.qty
+
+    def init(self, cr):
+        cr.execute('SELECT indexname FROM pg_indexes WHERE indexname = %s', ('stock_quant_product_location_index',))
+        if not cr.fetchone():
+            cr.execute('CREATE INDEX stock_quant_product_location_index ON stock_quant (product_id, location_id, company_id, qty, in_date, reservation_id)')
+
+    @api.multi
+    def unlink(self):
+        if not self.env.context.get('force_unlink'):
+            raise UserError(_('Under no circumstances should you delete or change quants yourselves!'))
+        return super(Quant, self).unlink()
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        ''' Overwrite the read_group in order to sum the function field 'inventory_value' in group by'''
+        # TDE NOTE: WHAAAAT ???
+        res = super(Quant, self).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
+        # if 'inventory_value' in fields:
+        #     for line in res:
+        #         if '__domain' in line:
+        #             lines = self.search(cr, uid, line['__domain'], context=context)
+        #             inv_value = 0.0
+        #             for line2 in self.browse(cr, uid, lines, context=context):
+        #                 inv_value += line2.inventory_value
+        #             line['inventory_value'] = inv_value
+        return res
+
+    @api.multi
+    def action_view_quant_history(self):
+        ''' Returns an action that display the history of the quant, which
+        mean all the stock moves that lead to this quant creation with this
+        quant quantity. '''
+        action = self.env.ref('stock', 'stock_move_action')
+        result = action.read()
+        move_ids = self.mapped('history_ids').ids
+        result['domain'] = [('id', 'in', ','.join(map(str, move_ids)))]
```

Comment:
That looks weird, why would the ids need to be joined into a comma-separated string? /cc @ged-odoo